austation/code/modules/mining/minebot.dm

```diff
@@ -0,0 +1,95 @@
+#define MINEDRONE_COLLECT 1
+#define MINEDRONE_ATTACK 2
+
+/mob/living/simple_animal/hostile/mining_drone
+	var/beacons = 15
+	var/default_hatmask
+	var/obj/item/t_scanner/adv_mining_scanner/lesser/scanner //there is no code to turn this scanner off.
+
+/mob/living/simple_animal/hostile/mining_drone/Initialize()
+	. = ..()
+	weather_immunities += "ash" // no damage from ash storms
+	stored_gun.overheat_time = 10
+
+	scanner = new(src)
+	scanner.toggle_on()
+	var/datum/action/innate/minedrone/marker_beacon/beacon_action = new()
+	beacon_action.Grant(src)
+
+/datum/action/innate/minedrone/marker_beacon
+	name = "Drop Marker"
+	button_icon_state = "mech_zoom_off"
+
+/datum/action/innate/minedrone/marker_beacon/Activate()
+	var/mob/living/simple_animal/hostile/mining_drone/M = owner
```

Comment:
you're typecasting but not actually checking the type? add an istype() check somewhere, else just use `?.` and don't worry about typecasting at all.

Reply:
now checking for istype()